Lib/asynchat.py

```diff
@@ -122,7 +122,7 @@ def handle_read(self):
             return
 
         if isinstance(data, str) and self.use_encoding:
-            data = bytes(str, self.encoding)
+            data = bytes(data, self.encoding)
```

Comment:
I think we need an issue on the bug tracker for discussing. GitHub is used only for discussing technical details of a patch.